src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -197,6 +200,19 @@ bool hasNamespaceSuppression(INamespaceSymbol namespaceSymbol, bool inImmediatel
 
                 return false;
             }
+
+            static bool isCompilerDiagnostic(Diagnostic diagnostic)
+            {
+                foreach (var customTag in diagnostic.CustomTags)
```

Comment:
how does IReadOnlyList not have a .Contains method...

Reply:
Sure, will just inline.

Reply:
Agreed, we should use `System.Linq.Enumerable.Contains()` or some equivalent.

Reply:
I would prefer to use a `for` loop with indexing to avoid enumerator allocations.

Reply:
Thanks, I added an extension method with the implementation using a for loop.